includes/modules/pages/popup_image/header_php.php

```diff
@@ -38,12 +38,12 @@
   $products_image = $products_values->fields['products_image'];
 
   //auto replace with defined missing image
-  if ($products_image == '' and PRODUCTS_IMAGE_NO_IMAGE_STATUS == '1') {
+  if ($products_image === '' && PRODUCTS_IMAGE_NO_IMAGE_STATUS === '1') {
     $products_image = PRODUCTS_IMAGE_NO_IMAGE;
   }
 
-  $products_image_extension = substr($products_image, strrpos($products_image, '.'));
-  $products_image_base = preg_replace('|'.$products_image_extension.'$|', '', $products_image);
+  $products_image_extension = '.' . pathinfo($products_image, PATHINFO_EXTENSION);
+  $products_image_base = str_replace($products_image_extension, '', $products_image);
```

Comment:
Changing from `preg_replace('...$')` to `str_replace()` bypasses the pattern of only stripping the extension off the end of the filename.

eg: a filename of `product_gift_image1.gif` would now use an incorrect base of `product_t_image1`

Reply:
Actually, the filename would need to be `product.gift_image1.gif`, but point taken.

Reply:
true